xayn-ai/src/coi/relevance.rs

```diff
@@ -251,6 +257,20 @@ impl RelevanceMap {
             _ => None,
         }
     }
+
+    pub(super) fn insert_cleaned_if_empty(&mut self) {
+        if self.relevance_to_key_phrase.is_empty() {
+            let to_insert = std::mem::take(&mut self.cleaned);
+
+            for (coi_id, value) in to_insert.into_iter() {
+                // we need to insert in reverse order to keep the order in which
+                // the key phrases were originally
```

Comment:
i double checked this, we could update the comment and point to the exact reason why we need to reverse. in `RelevanceMap::select_top_key_phrases()` we collect and sort the key phrases in descending order wrt the penalized relevances, then the selected key phrases are cleaned in that reversed order.

it's also important that this simple reverse only works if we keep up the invariant that the `penalty` must be sorted decreasingly. if we ever need to change that, then `cleaned` needs to also track the penalized relevances and reverse the insert order wrt them.